onchain/cairo/games/src/pixel/art_peace.cairo

```diff
@@ -508,30 +552,46 @@ pub mod ArtPeace {
         }
 
 
-        fn _place_shield(
-            ref self: ContractState, pos: u128, color: u8, now: u64, metadata: MetadataPixel
-        ) {
-            // self._check_shield_ok(pos, color);
+        fn _place_shield(ref self: ContractState, pos: u128,) {
+            let caller = get_caller_address();
 
-            // place_pixel_inner(ref self, pos, color);
-            // self._place_pixel_inner(pos, color);
-
-            // TODO: let pixel = PixelState { color, pos,  owner: caller,
-            // timestamp:get_block_timestamp()};
-            // TODO: self.canvas.write(pos, pixel);
-            // TODO: self.last_placed_pixel_shield.write(pos, pixel);
-            let caller = starknet::get_caller_address();
-            let day = self.day_index.read();
+            // check if pixel shield is activated
+            assert(self.is_shield_pixel_activated.read(), 'shield not activated');
 
-            self.last_placed_time.entry(caller).write(now);
+            // check owner of last placed pixel
+            let pixel_placed = self.last_placed_pixel.read(pos);
+            assert(pixel_placed.owner == caller, 'wrong pixel owner');
+
+            // check if pos has shield
+            assert(!self._has_shield(pos), 'pixel under shield');
+
+            // buy shield and transfer amount
+            let shield_type = self.shield_type.read();
+            let shield_params = self.admin_shield_params.entry(shield_type.clone()).read();
+            let amount_paid = shield_params.amount_to_paid;
+            //TODO
+            // IERC20Dispatcher { contract_address: 1.try_into().unwrap() }
+            //     .transfer_from(caller, 1.try_into().unwrap(), amount_paid.clone());
+
+            // set shield
+            let shield = PixelShield {
+                pos,
+                timestamp: get_block_timestamp(),
+                until: shield_params.until,
```

Comment:
If the user buy a shield when ShieldType = BuyTime:

An user need to choose how many time he want to buy this shield for his pixel